File: tests/fixtures/java/AnonymousIteratorCapturesField.java
import java.util.NoSuchElementException;

public class AnonymousIteratorCapturesField {
    private final String[] values;
    private int offset;

    public AnonymousIteratorCapturesField(String[] values, int offset) {
        this.values = values;
        this.offset = offset;
    }

    public Object iterator() {
        return new Iterator<String>() {
            private int index = offset;

            @Override
            public boolean hasNext() {
                return index < values.length;
            }

            @Override
            public String next() {
                if (index >= values.length) {
                    throw new NoSuchElementException();
                }
                return values[index++];
            }
        };
    }
}

File: tests/fixtures/python/AnonymousIteratorCapturesField.py
from __future__ import annotations


class AnonymousIteratorCapturesField:

    def __init__(self, values: list[str], offset: int) -> None:
        self.values = values
        self.offset = offset

    def iterator(self) -> object:

        _outer_self = self

        class _J2pyAnonymous1:
            def __init__(self):
                self.index: int = _outer_self.offset

            # @Override
            def has_next(self) -> bool:
                return self.index < len(_outer_self.values)

            # @Override
            def next_(self) -> str:
                if self.index >= len(_outer_self.values):
                    raise StopIteration()
                self.index += 1
                return _outer_self.values[(self.index - 1)]
        return _J2pyAnonymous1()
